src/Properties/SchemaAggregator.php

```diff
@@ -160,6 +160,43 @@ private function processColumnUpdates(string $tableName, string $argName, array
                     $firstArg = $firstMethodCall->args[0]->value ?? null;
                     $secondArg = $firstMethodCall->args[1]->value ?? null;
 
+                    if ($firstMethodCall->name->name === 'foreignIdFor') {
+                        if ($firstArg instanceof PhpParser\Node\Expr\ClassConstFetch
+                            && $firstArg->class instanceof PhpParser\Node\Name
+                        ) {
+                            $modelClass = $firstArg->class->toCodeString();
+                        } elseif ($firstArg instanceof PhpParser\Node\Scalar\String_) {
+                            $modelClass = $firstArg->value;
+                        } else {
+                            continue;
+                        }
+
+                        if (! class_exists($modelClass)) {
+                            continue;
+                        }
+
+                        try {
+                            $reflect = new \ReflectionClass($modelClass);
+
+                            /** @var \Illuminate\Database\Eloquent\Model $model */
+                            $model = $reflect->newInstanceWithoutConstructor();
+
+                            $columnName = $model->getForeignKey();
+                            $columnType = $model->getKeyType() === 'int' && $model->getIncrementing()
+                                ? 'int'
+                                : 'string';
+
+                            if ($secondArg instanceof PhpParser\Node\Scalar\String_) {
+                                $columnName = $secondArg->value;
+                            }
```

Comment:
Can we replace this code here with `$columnName = Str::snake(class_basename($modelClass)) . '_id'` ? I think this will cover most of the cases, and we would avoid using reflection.

Reply:
this was based on similar reflection in https://github.com/nunomaduro/larastan/blob/e35de81506c5208ad8cf923f6dae0034e51f4615/src/Properties/ModelPropertyExtension.php#L74-L79